### PR TITLE
Unit test and CI, Base setup: - added `JUnit5`; - setup `Spek 2`; - `…

### PR DESCRIPTION
**Pull request by RingerJK**
…travis` setup.

Refactored: `kotlin` version 1.3.21 -> 1.3.30.
TotalByteCount.kt: refactored and cover by tests.

### CODE REVIEW

File: MapboxVision/src/main/java/com/mapbox/vision/utils/system/SystemTime.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.mapbox.vision.utils.system
+
+interface SystemTime {
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/utils/system/SystemTime.kt]: Plain `System`, not `SystemTime`?

Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/utils/system/SystemTime.kt]: System already exist and has pretty huge API, I suggest to use small interface for Time only

**Comment by RingerJK** [comment]
@yunikkk ready for review

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt [hunk 2]
```diff
@@ -12,45 +16,51 @@ internal interface TotalBytesCounter {
...
-            private val sessionLengthMillis: Long = SESSION_LENGTH_MILLIS,
-            private val sessionMaxBytes: Long = SESSION_MAX_BYTES
+        private val sessionLengthMillis: Long = SESSION_LENGTH_MILLIS,
+        private val sessionMaxBytes: Long = SESSION_MAX_BYTES,
+        private val systemTime: SystemTime = SystemTime.Impl
     ) : TotalBytesCounter {
 
         companion object {
-            private const val SESSION_LENGTH_MILLIS = 60 * 60 * 1000L // one hour
+            private val SESSION_LENGTH_MILLIS by lazy(LazyThreadSafetyMode.NONE) { TimeUnit.HOURS.toMillis(1) }
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: Do we really need lazy here?

Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: when we set this value through constructor we don't need this value.

File: MapboxVision/src/main/java/com/mapbox/vision/utils/system/SystemTime.kt (hunk 1 above)
Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/utils/system/SystemTime.kt]: Well, we might implement only those methods from System, that we actually need. 

What about renaming class to `interface Time { fun millis() }`  ?
Usage will become shorter.

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt (hunk 2 above)
Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: Right, but is it worth to create additional objects for this cases? I believe 1-time multiplication will be even faster then property access overhead

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt [hunk 3]
```diff
@@ -12,45 +16,51 @@ internal interface TotalBytesCounter {
...
+            val timestamp = systemTime.currentTimeMillis()
 
             return when {
-                sessionStartMillis + sessionLengthMillis < timestamp -> {
+                sessionStartMillis + sessionLengthMillis <= timestamp -> {
                     sessionStartMillis = timestamp
-                    bytesSentPerSession = bytes
-                    true
+                    bytesSentPerSession = 0
+                    return if (fitInLimit(bytes)) {
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: Good catch! Should we add ticket to handle such situations?

Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: Yep, I think so. A delgate/listener should notify user about it

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt (hunk 2 above)
Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: yes, but I'd lite to use `TimeUnit` util class for time. It useful and clean. btw `SESSION_LENGTH_MILLIS ` doesn't call many times

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt (hunk 3 above)
Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: Do you plan to add it in this PR? Or create separate ticket to handle this case?

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt (hunk 2 above)
Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: So can we go with `private val ... = TimeUnit.HOURS.toMillis(1)`?

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt (hunk 3 above)
Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: I think this PR is about small refactoring, setup test environment and test introducing. I'd like to create separate ticket. I'll make it.

File: MapboxVision/src/main/java/com/mapbox/vision/utils/system/SystemTime.kt (hunk 1 above)
Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/utils/system/SystemTime.kt]: done

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt (hunk 2 above)
Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: what are pros of this change?

Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: let's discuss it offline :)

File: MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt (hunk 3 above)
Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/telemetry/TotalBytesCounter.kt]: Sure, great!

**Pull request merged**